Configure kernel build with IP_NF_TARGET_TTL

Moris-Shalon opened the pull request:
Please, add "TTL target support" module to kernel config. This module is required to bypass cellular operator's restrictions on Wi-Fi tethering. It makes possible to change TTL of all incoming and outgoing packets, so tracking Wi-Fi tethering becomes harder for cellular operator.

Avoiding restrictions on LTE -> Wi-Fi/Bluetooth/USB tethering becomes easy with this module, using simple script through Magisk:

\#!/system/bin/sh
iptables -t mangle -A POSTROUTING -o rmnet_data+ -j TTL --ttl-set 64
iptables -t mangle -I PREROUTING -i rmnet_data+ -m ttl --ttl 1 -j TTL --ttl-inc 4
ip6tables -t mangle -A POSTROUTING ! -o rmnet_data+ -j HL --hl-set 64
ip6tables -t mangle -I PREROUTING ! -i rmnet_data+ -m hl --hl 1 -j HL --hl-inc 4

\# Check that everything is working
\#iptables -t mangle -L -n | grep -i TTL

Moris-Shalon committed:
Configure kernel build with IP_NF_TARGET_TTL

Add "TTL target support" module to kernel config. This module is required to bypass cellular operator's restrictions on Wi-Fi tethering. It makes possible to change TTL of all incoming and outgoing packets, so tracking Wi-Fi tethering becomes harder for cellular operator.

Avoiding restrictions on LTE -> Wi-Fi/Bluetooth/USB tethering becomes easy with this module, using simple script through Magisk:

\#!/system/bin/sh
iptables -t mangle -A POSTROUTING -o rmnet_data+ -j TTL --ttl-set 64
iptables -t mangle -I PREROUTING -i rmnet_data+ -m ttl --ttl 1 -j TTL --ttl-inc 4
ip6tables -t mangle -A POSTROUTING ! -o rmnet_data+ -j HL --hl-set 64
ip6tables -t mangle -I PREROUTING ! -i rmnet_data+ -m hl --hl 1 -j HL --hl-inc 4

\# Check that everything is working
\#iptables -t mangle -L -n | grep -i TTL

Moris-Shalon committed:
Configure kernel build with IP_NF_TARGET_TTL

Please ,add "TTL target support" module to kernel config. This module is required to bypass cellular operator's restrictions on Wi-Fi tethering. It makes possible to change TTL of all incoming and outgoing packets, so tracking Wi-Fi tethering becomes harder for cellular operator.

Avoiding restrictions on LTE -> Wi-Fi/Bluetooth/USB tethering becomes easy with this module, using simple script through Magisk:

\#!/system/bin/sh
iptables -t mangle -A POSTROUTING -o rmnet_data+ -j TTL --ttl-set 64
iptables -t mangle -I PREROUTING -i rmnet_data+ -m ttl --ttl 1 -j TTL --ttl-inc 4
ip6tables -t mangle -A POSTROUTING ! -o rmnet_data+ -j HL --hl-set 64
ip6tables -t mangle -I PREROUTING ! -i rmnet_data+ -m hl --hl 1 -j HL --hl-inc 4

\# Check that everything is working
\#iptables -t mangle -L -n | grep -i TTL

Moris-Shalon committed:
Merge pull request #1 from ZChuckMoris/patch-2

Configure kernel build with IP_NF_TARGET_TTL

Moris-Shalon committed:
Merge pull request #2 from ZChuckMoris/patch-1

Configure kernel build with IP_NF_TARGET_TTL

UtsavBalar1231 commented:
Noted and will approve the support
But i will not merge this pull as we do not edit 
arch/arm64/configs/*_defconfig manually.
These configs are auto generated through scripts/update_defconfig


Instead you can add the config entry inside arch/arm64/configs/vendor/xiaomi/kona-custom.config

Moris-Shalon committed:
Configure kernel build with IP_NF_TARGET_TTL

Add "TTL target support" module to kernel config. This module is required to bypass cellular operator's restrictions on Wi-Fi tethering. It makes possible to change TTL of all incoming and outgoing packets, so tracking Wi-Fi tethering becomes harder for cellular operator.

Avoiding restrictions on LTE -> Wi-Fi/Bluetooth/USB tethering becomes easy with this module, using simple script through Magisk:

\#!/system/bin/sh
iptables -t mangle -A POSTROUTING -o rmnet_data+ -j TTL --ttl-set 64
iptables -t mangle -I PREROUTING -i rmnet_data+ -m ttl --ttl 1 -j TTL --ttl-inc 4
ip6tables -t mangle -A POSTROUTING ! -o rmnet_data+ -j HL --hl-set 64
ip6tables -t mangle -I PREROUTING ! -i rmnet_data+ -m hl --hl 1 -j HL --hl-inc 4

\# Check that everything is working
\#iptables -t mangle -L -n | grep -i TTL

Moris-Shalon committed:
Configure kernel build with IP_NF_TARGET_TTL

Add "TTL target support" module to kernel config. This module is required to bypass cellular operator's restrictions on Wi-Fi tethering. It makes possible to change TTL of all incoming and outgoing packets, so tracking Wi-Fi tethering becomes harder for cellular operator.

Avoiding restrictions on LTE -> Wi-Fi/Bluetooth/USB tethering becomes easy with this module, using simple script through Magisk:

\#!/system/bin/sh
iptables -t mangle -A POSTROUTING -o rmnet_data+ -j TTL --ttl-set 64
iptables -t mangle -I PREROUTING -i rmnet_data+ -m ttl --ttl 1 -j TTL --ttl-inc 4
ip6tables -t mangle -A POSTROUTING ! -o rmnet_data+ -j HL --hl-set 64
ip6tables -t mangle -I PREROUTING ! -i rmnet_data+ -m hl --hl 1 -j HL --hl-inc 4

\# Check that everything is working
\#iptables -t mangle -L -n | grep -i TTL"
